test/reply-error.test.js

```diff
@@ -396,3 +397,35 @@ test('should throw an error if the custom serializer does not serialize the payl
     t.fail('should not be called')
   })
 })
+
+// Issue 2078 https://github.com/fastify/fastify/issues/2078
+// Supported error code list: http://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml
+test('should throw error if error code is invalid', async t => {
+  const invalidErrorCodes = [
+    undefined,
+    null,
+    'error_code',
+    700 // out of the 100-600 range
+  ]
+  const fastify = Fastify()
+  const inject = promisify(fastify.inject).bind(fastify)
```

Comment:
there is no need to promisify inject, it returns a promise already. Maybe you might want to update the docs as well for that in another PR.

Here is the full API: https://github.com/fastify/light-my-request